src/pool/srv_pool.c

```diff
@@ -6158,25 +6208,38 @@ ds_pool_update_handler(crt_rpc_t *rpc)
 }
 
 static int
-pool_svc_exclude_rank(struct pool_svc *svc, d_rank_t rank)
+pool_svc_change_rank(struct pool_svc *svc, d_rank_t rank, crt_opcode_t opc)
 {
 	struct pool_target_addr_list	list;
 	struct pool_target_addr_list	inval_list_out = { 0 };
 	struct pool_target_addr		tgt_rank;
 	uint32_t			map_version = 0;
-	int				rc;
+	int				rc = 0;
+	bool				exclude_rank = false;
 
+	D_ASSERTF(opc == POOL_REINT || opc == POOL_EXCLUDE, " opc = %u\n", opc);
 	tgt_rank.pta_rank = rank;
 	tgt_rank.pta_target = -1;
 	list.pta_number = 1;
 	list.pta_addrs = &tgt_rank;
 
-	rc = pool_svc_update_map(svc, POOL_EXCLUDE, true /* exclude_rank */,
+	switch (opc) {
+	case POOL_REINT:
+		exclude_rank = false;
+		break;
+	case POOL_EXCLUDE:
+		exclude_rank = true;
+		break;
+	default:
+		break;
+	}
+
+	rc = pool_svc_update_map(svc, opc, exclude_rank /* exclude_rank */,
```

Comment:
exclude_rank and opc seems duplicate parameters?

Reply:
well， exclude_rank here  is to distinguish between swim detection and command. The swim detection will mark rank down, but the command will not mark rank down